# SDAS: Stereo-seq Data Analysis Solution
SDAS software is a comprehensive bioinformatics toolkit designed for the analysis of Stereo-seq transcriptomic data, covering all key steps in data interpretation.
## Features
9 Functional Modules with 14 Algorithms
  
| Functional Module               | Algorithm/Method        | Environment         | Key package                  |
|:--------------------------------|:-------------------------|:--------------------|:------------------------------|
| Data process                    | -                       | python 3.10; R 4.3.3| schard 0.0.1, seurat 5.1.0, scanpy 1.10.4 |
| Cell type annotation            | Spotlight               | R 4.3.3              | spotlight 1.6.3               |
|                                 | cell2location           | python 3.10          | cell2location 0.1.4           |
|                                 | RCTD                    | R 4.3.3              | r-spacexr 2.2.1               |
| Spatial domain                  | GraphST                 | python 3.8           | GraphST 1.1.1                 |
| CNV analysis                    | inferCNV                | R 4.2.3              | infercnv 1.14.2               |
| Spatially co-expressed genes    | hotspot                 | python 3.10          | nest-analysis 1.0.4           |
|                                 | nest                    | python 3.10          | hotspotsc 1.1.1               |
| DEG                             | scanpy                  | python 3.10          | scanpy 1.10.4                 |
| Gene set enrichment analysis    | gsea                    | python 3.10          | gseapy 1.1.5             |
|                                 | enrichr                 | python 3.10          | gseapy 1.1.5                  |
|                                 | prerank                 | python 3.10          | gseapy 1.1.5                              |
|                                 | gsva                    | python 3.10          | gseapy 1.1.5                             |
| Cell-cell communication analysis| cellchat v2             | R 4.3.3              | CellChat 2.1.2                |
| Trajectory analysis             | monocle3                | R 4.3.3              | monocle3 1.3.1                |
## Inputs
- h5ad Files Generated by SAW count and SAW convert gef2h5ad (with fixed bin size)
- h5mu Files Generated by SAW aggr
## Outputs 
- h5ad, rds, csv or txt File
- each functional module also generates corresponding figures in PNG and PDF format.
## System Requirements
The SDAS software package is unpacked and installed on a Linux system. The computational environment must meet the following basic requirements:  
- 8-core Intel or AMD processor (>24 cores recommended)  
- 128GB RAM (>256GB recommended)  
- 200G free disk space or higher  
- 64-bit CentOS/RedHat 7.8 or Ubuntu 20.04  
## Software Download
BGI Cloud Disk: https://bgipan.genomics.cn/#/link/29ateK2tAQYXGhetPYk7  code: vERA  
GitHub:  
```bash
git clone https://github.com/STOmics/SDAS.git
```
Note: The repository does not include .tar.gz file or test data.
## Unpack
```bash
tar -xzf SDAS_beta.tar.gz
cd SDAS_beta/anno
./conda-unpack
```
## Test run
```bash
./SDAS -h
```
<pre>
SDAS functions:
    dataProcess                              Prerocess data, include mergeAdata, h5ad2rds, h5mu2h5ad, printAdataInfo and subsetAdata                            
    cellAnnotation                           Annotate celltype of spatial transcriptomics data                              
    coexpress                                Identify spatially coexpressed gene modules in spatial transcriptomics data                      
    trajectory                               Reconstruct cell trajectories from spatial transcriptomics or scRNA-Seq data                            
    CCI                                      Infer cell-cell communication in spatial transcriptomics or scRNA-Seq data               
    infercnv                                 Detect copy number variations in spatial transcriptomic or scRNA-Seq data                           
    spatialDomain                            Identify spatial domain of spatial transcriptomics data                        
    DEG                                      Identify differentially expressed genes of spatial transcriptomics or scRNA-Seq data        
    geneSetEnrichment                        Analyze functional enrichment of gene sets in spatial transcriptomics or scRNA-Seq data 
	</pre>

## SDAS Material Overview
<pre>
├── SDAS_beta.tar.gz
├── Documents
│   └── SDAS_beta_user_manual.pdf
├── Scripts
│   ├── pipelines
│   └── quick_start
└── Test_data
│   ├── single_slice
│   └── multiple_slices
└── Application_cases
     ├── Application_scripts
     ├── Application_test_data
     │   ├── single_cell.h5ad
     │   └── stereo_bin100.h5ad
     └── NC2024_paper_bioinformatics_analysis_withSDAS.pdf
</pre>
